### PR TITLE
Feature:Lesson12

### PR DESCRIPTION
**Pull request by nyattsu72**
## Outline
Worked on Javascript lesson12.
[Lesson12](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#12)

[codeSandbox](https://codesandbox.io/s/wizardly-frost-h3xtr?file=/css/styles.css)

## Task
- [x]  The menu is called when the button is clicked.
 Add code to execute callMenuContents function with addEventListener

- [x]  Prevent duplicate menus from appearing on second click
 If the menu is displayed, delete the menu.
 Start reading after erasing.
- [x] Change the loading to load on every click

## Places you want to check

- When I click the button, the menu grows infinitely, so I tried to initialize the list. Is this the method?
(I’m commenting.)

### CODE REVIEW

File: lesson12/js/main.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function hideLoadingImg() {
+	const loading = document.getElementById('js-loading');
+	loading.remove();
+}
+
+function createMenuList(data) {
+	const fragment = document.createDocumentFragment();
+	menuLists.innerHTML = '';
```
Comment (nyattsu72) [inline lesson12/js/main.js]: When I click the button, the menu grows infinitely, so I tried to initialize the list. Is this the method?

Comment (kenmori) [reply lesson12/js/main.js]: ok,
improve your code

```js
function removeInnerHTML(element){
 element.innerHTML = "";
}

function createMenuList(data) {
	const fragment = document.createDocumentFragment();
	removeInnerHTML(menuLists) // 何度もボタン押下した際に積まれていくので都度初期化している

Comment (kenmori) [reply lesson12/js/main.js]: Another solution is..
you can pass third argument for addEventListener as option. once,
User can click the button only once.
add `disabled true` for button style.
```js
clickBtn.addEventListener(
  "click",
  () => {
    callMenuContents();
  },
  { once: true }
);
```

**Comment by kenmori** [comment]
Approve

File: lesson12/js/main.js (hunk 1 above)
Comment (nyattsu72) [reply lesson12/js/main.js]: Thank you for review!

I implemented only one click received in the review with the following PR.
[once Click ver](https://github.com/nyattsu72/hanson-frontend-cource/pull/16)
May I ask you to confirm?

**Pull request merged**